Fix #3135:  Switch from Static Image Resources By Using Existing Font Awesome Icons

shaz13 opened the pull request:
As issued in #3135. Font awesome icons are adopted in this PR. I have further added required styling to pad them in-line accordingly.

**NOTE:** I haven't deleted the existing image resources in this commit. After getting initial review over font awesome, colors and screenreader, would push another commit deleting the image resources located at 
`/oppia/assets/images/social`

**OLD**
--------------------------------------------------------------------------------------------------------------------------
> **Before PR**
> 
> ![before](https://cloud.githubusercontent.com/assets/24438869/23832529/04784c68-075c-11e7-965c-df37489abad7.jpg)
> 
> 
> **Result If: after merging PR**
> ![fonticonss](https://cloud.githubusercontent.com/assets/24438869/23832509/8de60f5e-075b-11e7-9c66-e2fc217b899a.jpg)

----------------------------------------------------------------------------------------------------------------------------
**AFTER NEW COMMIT**

![cropped sign](https://cloud.githubusercontent.com/assets/24438869/23937263/930af198-097c-11e7-8c31-975ca221899b.jpg)

![fontawesome](https://cloud.githubusercontent.com/assets/24438869/23937264/930ca006-097c-11e7-8280-7403bdf79be7.jpg)

seanlip commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -1,21 +1,36 @@
 <script type="text/ng-template" id="components/socialButtons">
+<style type="text/css">
+  
+.fa {
```
This is defined too broadly -- it hits all the other .fa classes in the application, not just these ones. (Please grep for the locations of the others and verify that they are unaffected before submitting the PR.)

Also, why do you need it? If you need to style the elements in the footer then add a custom class instead that is prefixed with "oppia-" so we know it's one of ours.

seanlip commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -1,21 +1,36 @@
     </div>
     <a href="https://plus.google.com/109898456505810251700/about" target="_blank">
-      <img ng-src="<[getStaticImageUrl('/social/gplus.png')]>" alt="Google+">
+      <i class="fa fa-google-plus-square" aria-hidden="true"></i>
```
@anthkris PTAL at the aria-hidden; is that the correct thing to do?

anthkris replied:
No, I don't think that's necessary and I would remove it from here and below. However, what is necessary is to include the oppia accessibility fix for icons. 

@shaz13 In order to ensure that icons are screenreader accessible, we need to include a label. The fix that we have implemented is to use a span with a special class that will allow the screenreader to read labels for icons.
It looks like this:
`<span class="oppia-icon-accessibility-label">Delete this list entry</span>`

And you can see more about the implementation in PR #3118 

So when you switch out a graphic with an alt attribute for an icon, you will need to use a span like the one above with appropriate text (which should be taken from the alt attribute). The span should go directly beneath the icon. Does that make sense?

markhalpin commented:
To be honest, I prefer the little splashes of color that the original icons give us. What's the motivation for making these monochromatic?

shaz13 commented:
@anthkris  @seanlip Pushed a new commit. Made class more specific and removed and removed `aria-hidden` as suggested in the review. Apologies for not making it so in the first place. Am kind of learning stuff.

seanlip commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -1,21 +1,36 @@
 <script type="text/ng-template" id="components/socialButtons">
+<style type="text/css">
```
Please look at existing directives (such as [this one](https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/pages/library/activity_tiles_infinity_grid_directive.html)) and carefully follow their structure, indentation and formatting. Consistency of code style matters in this codebase.

seanlip commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -1,21 +1,36 @@
+<style type="text/css">
+  
+i.oppia-share-btn-fa {
+    display: inline-block;
```
Alphabetize. Please read the [coding style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide#css) if you haven't already, and ensure that your code conforms to it.

shaz13 replied:
@markhalpin That would look cool. Shall I make separate classes for them? So as to include the color to specific icon

seanlip replied:
@shaz13 I think you're commenting in the wrong place? The comment you are replying to was made by me, not by @markhalpin.

shaz13 replied:
Undoing my comment... 
@seanlip  
I alphabetized the properties. Kindly look whether they are according to oppia styling standards.

seanlip replied:
No, I'm afraid they are not. Please pay attention to indentation, and the position of the style tag in the file. See the [previous comment](https://github.com/oppia/oppia/pull/3193#discussion_r105592255) I made, and compare the file you are editing to the file mentioned in that comment.

shaz13 replied:
@seanlip Sure Sir, will correct them right away. And [ coding style guide ](https://github.com/oppia/oppia/wiki/Coding-style-guide#css) reads .. 

 For directives, include the CSS in the directive template file, similar to what we do in `core/templates/dev/head/components/exploration_summary_tile_directive.html`

But there is **no such file**, which made me unfamiliar with styling the code base.
Looks like the file is moved to  `summary_tile` folder, and right path is `core/templates/dev/head/components/summary_tile/exploration_summary_tile_directive.html`. 

However, the file you posted in comments give the better idea of the styling guide than former. 
`core/templates/dev/head/pages/library/activity_tiles_infinity_grid_directive.html`

Thank you ! .

shaz13 replied:
Pushed the another position and commit again.Kindly, do review it. Hope this time its all set.

seanlip replied:
Ah yes, the file was moved. I have fixed the link, thanks for reporting. (That said, the thing to do in that case, in addition to reporting it, is to look around for other directive files to get a sense of what the existing styling requirements are.)

Also, please remember to address the point in my top-level comment below (about the tradeoff between colors and speed).

seanlip commented:
Actually, let me help with the reply to @markhalpin, so that he has context.

@markhalpin, the previous images were being loaded as separate files, which increases bandwidth requirements a bit and delays the loading of the page. I think the aim of this PR is to replace them with an image font derived from [here](http://fontawesome.io/icons/), in order to remove the need for the additional loads. However, that font appears to be monochromatic and so we lose the color splashes.

So I think the tradeoff here is between colors and speed, which leads to two questions:
- @shaz13: how much time does using the fonts save, in practice? 
- @markhalpin: how bad is it without the colors? Do you think this justifies the (slight) decrease in speed needed to load the page?

Thanks!

seanlip commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -6,16 +6,30 @@
 </script>
+
+<style>
+  i.oppia-share-btn-fa {
```
I suggest `.oppia-footer-social i.oppia-share-btn-fa`, to scope it to this file.

seanlip commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -6,16 +6,30 @@
+
+<style>
+  i.oppia-share-btn-fa {
+   	display: inline-block;
```
The indentation is wrong. You should be using spaces, not tabs. Did you run the presubmit checks locally before pushing?

seanlip commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -6,16 +6,30 @@
+   	-moz-osx-font-smoothing: grayscale;
+    -webkit-font-smoothing: antialiased;
+  }
+</style>
```
End file with EOF. This should have been caught in the presubmit checks.

shaz13 replied:
I didn't run presubmit checks. Indentation in my sublime text looks so.. 
Don't know why commit shows wrong indent? 
Cause of the tabs thing? 
![indent](https://cloud.githubusercontent.com/assets/24438869/23847175/26ea90a8-07f7-11e7-9ef1-fbf1a0a77939.jpg)

himanshu-dixit commented:
I agree with @markhalpin, the main intention of filing the issue was to decrease no. of network requests but retaining original appearance of element.
For this @shaz13, i think you need to identify images that can be replaced using fontawesome, other than this sign in with google can also be replaced.
For above, maybe you can try enclosing it in block each with its own color (color of social network).

shaz13 commented:
@himanshu-dixit Sure. I was waiting for the review inorder to implement font awesome to **Sign In** button. Currently, I am new here, so doing it all in steps. Kindly review the code and latest commits. 
Thanks

jaredsilver commented:
One of the great features of using an icon font is that we can assign a color to the text, and it gets applied in the same way as applying color to normal text.

A middle-ground solution might be changing the colors of each FontAwesome icon to match the color of the corresponding image icon.

In addition, it might be worth seeing how it looks using the square version of each icon so everything is aligned nicely (`fa-google-plus-square`, `fa-twitter-square`, `fa-facebook-square`, `fa-youtube-square`).

shaz13 commented:
@himanshu-dixit  @jaredsilver Thanks for your review(s). I will upload a screenshot of coloured font awesome squared icons before my next pull. If it checks right according to the site's design we may proceed.

jaredsilver commented:
@shaz13 Great! You may find this website helpful for retrieving the proper colors: https://brandcolors.net

anthkris commented:
@shaz13 I just want to remind you to please implement the accessibility labels I described previously. Otherwise, screenreaders will not be able to properly read these icons to visually impaired users.

shaz13 commented:
@anthkris Sure Sir, I still remember it. Will see that review is applied in my coming push.

@jaredsilver @himanshu-dixit 
I have added color classes to font awesome, as well as applied square icons as suggested by @jaredsilver sir. 
This is how it looks; 

![coloredfontawesome](https://cloud.githubusercontent.com/assets/24438869/23864023/4d33d8d8-0837-11e7-9abf-b5ee641df941.jpg)

markhalpin commented:
Hi @shaz13 and @jaredsilver,

@shaz13, thanks for trying out that color solution. It was definitely worth taking a look. In my opinion, using the Font Awesome approach for the sharing icons isn't successful. When uncolored, they don't really have any visual punch. When colored, they aren't really a good look with the dark green background: they either don't contrast enough or they create a visual "jangle" that's not very pleasing. I'd recommend that we maintain our current visual approach to these particular icons (as in the "Before PR" look). Thanks!

shaz13 commented:
@markhalpin Yes, Sir. I felt the same, the coloured icons don't look punchy. 
Shall I try colors with non square icons?

seanlip commented:
Hi @shaz13, I would suggest just trying things so that the respondent (in this case Mark) can give you a yes/no answer. It's faster that way.

Also, with regards to this:

> Sure. I was waiting for the review inorder to implement font awesome to Sign In button. Currently, I am new here, so doing it all in steps.

... could I suggest doing the opposite, i.e. getting the PR to a place you're proud of and feel that it's ready to submit, before asking for feedback? We are getting a huge amount of PRs now, so it's kind of hard to do a lot of incremental reviewing.

Thanks!

shaz13 commented:
@seanlip Sure Sir. Will proceed as you say. 

Currently, I guess this PR is ending with either implementing font awesome or **not**. 
If its a no would close this PR. Thanks

seanlip commented:
I think this is a question for @himanshu-dixit, though it's great that we've done the experiments to validate part (2) of the original issue -- thanks @shaz13. 

@himanshu-dixit, could you be specific about what icons you want to replace? From @shaz13's experiments, does it look like this is feasible? It's fine if you want to keep the original issue open, but it needs to be more actionable.

Thanks!

himanshu-dixit commented:
The main though was to use available resource, in this way maybe not much network request can reduced. Also we can have more control on element, for eg - there's currently no hover effect. 
@shaz13 i think approach of making it work won't be that straight forward. You can try something like this,
and can position fa-facebook by using absolute.
<div class="fb-color">
	<i class="fa fa-facebook" aria-hidden="true"></i>
</div>

seanlip commented:
Hi @himanshu-dixit, what do you mean by "this"? Note that HTML in GitHub comments needs to be escaped using backticks, otherwise GitHub markdown just treats it as HTML.

shaz13 commented:
Pushed new commit. Done as @himanshu-dixit wished to adopt `div` to make it look like former footer icons as well as Sign In button. @anthkris Added spans for screen reader and deleted static resources after going through the codebase, kindly look into it.
@seanlip @jaredsilver Kindly review it. 
The New commit changes
![cropped sign](https://cloud.githubusercontent.com/assets/24438869/23937347/260dbb24-097d-11e7-9664-0c2e94671757.jpg)
![fontawesome](https://cloud.githubusercontent.com/assets/24438869/23937348/2612c4e8-097d-11e7-8ad2-8ac0c6acbece.jpg)

seanlip commented:
Thanks, @shaz13. Deferring my part of the review to @himanshu-dixit, but it strikes me that the Sign In button at the top right does not look the same as the existing one. It should look exactly the same.

himanshu-dixit commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -5,17 +5,63 @@
+    
+    <div class="oppia-footer-icons">
+
+        <div class="goole-plus-color">
```
google-plus-color

shaz13 replied:
Good catch. Will see into it right away

himanshu-dixit commented on core/templates/dev/head/components/top_navigation_bar/top_navigation_bar_directive.html:
```diff
@@ -164,7 +164,9 @@
             <a href style="padding: 0; width: 200px;" ng-click="onLoginButtonClicked()">
-              <img ng-src="<[getStaticImageUrl('/signin/Red-signin-Long-base-44dp.png')]>" alt="Sign in with Google">
+              <div class="signin-google-color">
+                <i class="oppia-signin fa fa-google-plus"></i><b>Sign in with Google</b>
```
oppia-google-icon, oppia-signin looks like representing whole button.

himanshu-dixit replied:
or maybe oppia-signin-google-icon

shaz13 commented:
@seanlip Hotfix push committed. I didn't include the seperator. In my opinion this looks clean. If seperator at Sign in button is mandatory then will push another commit. 
![hotfix](https://cloud.githubusercontent.com/assets/24438869/23941436/cffcfc06-098e-11e7-8b0f-5c7fbb4496b9.jpg)

seanlip commented:
Thanks for making the changes! I think it looks fine.

seanlip commented:
(but I still defer to @himanshu-dixit)

shaz13 commented:
@himanshu-dixit Code names are changed locally. This is how it looks when altering fb and youtube icons
![alter](https://cloud.githubusercontent.com/assets/24438869/23942454/610327b8-0992-11e7-8e77-7ffb959ef11f.jpg)

himanshu-dixit commented:
@shaz13 i really don't think icons need to change, the previous one looked good. Maybe in that you can try decreasing font-size of youtube icon and try to include separator in signin button.
P.S.-  we don't need to make design changes, after changing code try to compare changes with the current one.

shaz13 commented:
Code name and terminology updated as reviewed in the last commit by @himanshu-dixit . Made footer icons look like former ones. 
New commit changes. 
![modified](https://cloud.githubusercontent.com/assets/24438869/23945627/36539dd0-099d-11e7-9e78-36d4b2ec5d77.jpg)

Holded on separator code as @seanlip approved the hotfix

> Thanks for making the changes! I think it looks fine.

Adding of separator would add more `div` and CSS, and adding non-functional code could compromise speed. Could we skip the separator? It adds no function and results in less code without it.

himanshu-dixit commented:
@shaz13, that won't make much difference, it will be good idea to not remove the separator. Also can you check if youtube icon is horizontally centered? 
After these change and code review, this PR might ready to merge.

shaz13 commented:
@seanlip @himanshu-dixit 
Added seperator too. :smiley: 
![separator](https://cloud.githubusercontent.com/assets/24438869/24007807/b7632fd2-0a95-11e7-95ce-12df5cfce924.jpg)

himanshu-dixit commented:
That looks great 🙂, just a small change, singin with google in not centre
aligned.
On Thu, 16 Mar 2017 at 10:13 PM, Mohammad Shahebaz <notifications@github.com>
wrote:

> @seanlip <https://github.com/seanlip> @himanshu-dixit
> <https://github.com/himanshu-dixit>
> Added seperator too. 😃
> [image: separator]
> <https://cloud.githubusercontent.com/assets/24438869/24007807/b7632fd2-0a95-11e7-95ce-12df5cfce924.jpg>
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia/pull/3193#issuecomment-287117863>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGiDno9KCu9Do01zJDM7K4S-_GrChwgZks5rmWaegaJpZM4MajMY>
> .
>
-- 

Himanshu Dixit

shaz13 commented:
@himanshu-dixit Done :smiley: Pushed a new commit. Kindly review it.

anthkris commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -5,17 +5,73 @@
+
+        <div class="google-plus-color">
+            <a href="https://plus.google.com/109898456505810251700/about" target="_blank"><i class="fa fa-google-plus"></i>
+            <span class="oppia-icon-accessibility-label">Follow us at Google Plus</span>
```
@shaz13 For this and the ones below, simplify the messages in the label. Use the text from the alt tags. For example, for this one, the label should simply be "Google Plus."

shaz13 replied:
Latest commit is pushed. Removed `google-plus-color`

anthkris commented on core/templates/dev/head/components/top_navigation_bar/top_navigation_bar_directive.html:
```diff
@@ -176,5 +178,28 @@
       outline: auto 5px -webkit-focus-ring-color;
     }
+    .oppia-signin-google {
+     background: #dd4b39;
```
Nit: The indentation here and below appears to be wrong.

shaz13 replied:
Corrected

anthkris commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -5,17 +5,73 @@
+        margin-right: 6px;
+        padding: 5px 6px;
+    }
+    .fb-color i.fa.fa-facebook {
```
Not sure about these class names, as we normally namespace our CSS to prevent possible clashes with third-party styles.

shaz13 replied:
corrected

shaz13 commented:
@anthkris Made changes in class, hope its developer friendly now.

himanshu-dixit commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -8,25 +8,25 @@
     <div class="oppia-footer-icons">
 
-        <div class="google-plus-color">
+        <div class="Google Plus">
```
@shaz13 i think you misinterpreted the @anthkris review.

himanshu-dixit commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -38,40 +38,40 @@
+    margin-right: 6px;
+    padding: 5px 6px;
+  }
+  .Facebook i.fa.fa-facebook {
```
As @anthkris commented, don't use third party CSS classes such as fa-facebook. Instead add new class (maybe with oppia prefix and suitable name) and then define the property.

shaz13 replied:
So, `Google Plus` class should be `oppia google plus` and for font awesome `facebook oppia fa-facebook `. Will this suffuse? @himanshu-dixit

shaz13 replied:
Also, I amt sure why the build is failing when there is no backend code change. Is it cause of static image deletion? Kindly guide me if  miss something else

seanlip replied:
See https://github.com/oppia/oppia/wiki/If-your-build-fails for the e2e test error. It's probably a flake.

For the backend test error, see if you can reproduce this locally, then look at the test that is erroring. Some backend tests look at frontend templates.

shaz13 replied:
@seanlip  Thanks alot. Will try to produce it locally.

shaz13 replied:
@seanlip Sir, 
Found the [error](https://travis-ci.org/oppia/oppia/jobs/212068496#L6608) at line number 6608 in the logs. It says "ERROR:  test_signup_page_does_not_have_top_right_menu (core.controllers.profile_test.SignupTest)" as well as IndexError: Body contains bad string '>Sign in'

In adopting font awesome we removed **Sign In with Google Image** and replaced with this PRs' code causing this error.
What might be the process now?

seanlip replied:
This one is tricky. I think you can remove `>Sign in` from the response.mustcontain list; that should work. Note that you can run the backend tests locally before you push, in order to double check -- see the instructions on the wiki.

shaz13 commented:
@seanlip Modified the backend test, pushed a commit on it. Travis build is passed :smile:  Thanks!
@anthkris Made fa classes more specific.
Wating the review

himanshu-dixit commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -1,21 +1,77 @@
-      <img ng-src="<[getStaticImageUrl('/social/twitter.png')]>" alt="Twitter">
-    </a>
+    
+    <div class="oppia footer icons">
```
same here, 2 classes

shaz13 replied:
Explanation provided in next comment. Kindly check it

himanshu-dixit commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -1,21 +1,77 @@
+    
+    <div class="oppia footer icons">
+
+        <div class="Google Plus">
```
Why 2 classes?

shaz13 replied:
The first class hold the Google, Facebook, Youtube and twitter icons in place, as block.
The sub divs are individual sqaured boxes.

shaz13 replied:
`<div class="oppia footer social">` the top most block holds, the "Follow us" translations code and the footer icons block.

himanshu-dixit replied:
It's not what i'm talking about? There are 3 classes in first block element, and in second there are 2 classes.

shaz13 replied:
Oh, I got it now. That was modified from `google-plus-color `to `Google Plus` following the review comment of anthkris. And similar space in class terminology was adopted.

![comm](https://cloud.githubusercontent.com/assets/24438869/24329209/e186cfb0-121c-11e7-8eae-fc6dff84d2b0.jpg)

shaz13 replied:
Got it. 
Classes are changed in the latest commit and few changes from the previous reviews are added too. Kindly review the latest one

himanshu-dixit commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -1,21 +1,77 @@
+            </a>
+        </div>
+        
+        <div class="Twitter">
```
oppia-twitter-follow might be sort of relevant class.

shaz13 replied:
Done

himanshu-dixit commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -1,21 +1,77 @@
+        </div>
+        
+        <div class="Twitter">
+            <a href="https://twitter.com/oppiaorg" target="_blank"><i class="oppia fa fa-twitter"></i>
```
I don't think oppia in class is right? It might need to be more descriptive.

shaz13 replied:
`oppia-font-awesome-icons`? Would this be good?

shaz13 replied:
Done

himanshu-dixit commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -1,21 +1,77 @@
 <script type="text/ng-template" id="components/socialButtons">
-  <div class="oppia-footer-social">
+  <div class="oppia footer social">
```
oppia-footer-social

shaz13 replied:
Guess, this class was untouched till now. However pushed a commit.Done

himanshu-dixit commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -1,21 +1,77 @@
 </script>
+
+<style>
+  .oppia.footer.social .oppia-footer-icons {
```
same here, apart from that code looks good to me.

shaz13 replied:
Done

seanlip commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -5,17 +5,73 @@
+    
+    <div class="oppia-footer-icons">
+
+        <div class="oppia-googleplus-follow">
```
All indentation in this file should be by 2 spaces, not 4. Please follow surrounding conventions.

shaz13 replied:
Work in progress

seanlip commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -5,17 +5,73 @@
-      <img ng-src="<[getStaticImageUrl('/social/twitter.png')]>" alt="Twitter">
-    </a>
+    
+    <div class="oppia-footer-icons">
```
oppia-footer-icons is vague; let's call it oppia-footer-social-icons

shaz13 replied:
Work in progress

seanlip commented on core/templates/dev/head/components/top_navigation_bar/top_navigation_bar_directive.html:
```diff
@@ -176,5 +178,33 @@
+      padding-right: 3px;
+      vertical-align: middle;
+    }
+    .oppia-signin-google-seperator {
```
"separator" is misspelled

shaz13 replied:
Work in progress

seanlip commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -6,60 +6,59 @@
 
 <style>
-  .oppia-footer-social .oppia-footer-icons {
+  .oppia-footer-social .oppia-footer-social-icons{
```
Nit: add space before {

shaz13 replied:
added

seanlip commented on core/templates/dev/head/components/social_buttons/social_buttons_directive.html:
```diff
@@ -6,60 +6,59 @@
-        </div>
+      <div class="oppia-twitter-follow">
+        <a href="https://twitter.com/oppiaorg" target="_blank"><i class="oppia-follow-icons fa fa-twitter"></i>
+            <span class="oppia-icon-accessibility-label">Twitter</span>
```
Indentation here is still incorrect; ditto above.

Also, I suggest moving the "i" tag to its own line, above this one. Ditto in various places above.

shaz13 replied:
Can, I use auto indent from sublime?

seanlip replied:
You can use whatever indenting tool you like, as long as whatever shows on the "Files Changed" tab (in GitHub) looks correct.

shaz13 replied:
Okay. Kindly review the latest commit

seanlip commented on core/templates/dev/head/components/top_navigation_bar/top_navigation_bar_directive.html:
```diff
@@ -164,7 +164,9 @@
             <a href style="padding: 0; width: 200px;" ng-click="onLoginButtonClicked()">
-              <img ng-src="<[getStaticImageUrl('/signin/Red-signin-Long-base-44dp.png')]>" alt="Sign in with Google">
+              <div class="oppia-signin-google">
+                <i class="oppia-signin-google-icon fa fa-google-plus"></i><div class="oppia-signin-google-separator"></div><b>Sign in with Google</b>
```
The div tag should move to the next line.
The b tag should move to the line after that.

shaz13 commented:
Done @seanlip. Must say you got Eagle Vision :eagle:   at PR review.

seanlip commented:
OK, thanks @shaz13!

@himanshu-dixit @anthkris PTAL and approve if you're happy, so that this can be merged. Thanks!

himanshu-dixit commented:
@seanlip LGTM, another thing can be adding hover effect on these buttons, like button bgcolor to slightly dark on hover. But that can be done in new PR.

anthkris commented on core/templates/dev/head/components/top_navigation_bar/top_navigation_bar_directive.html:
```diff
@@ -164,7 +164,11 @@
             <a href style="padding: 0; width: 200px;" ng-click="onLoginButtonClicked()">
-              <img ng-src="<[getStaticImageUrl('/signin/Red-signin-Long-base-44dp.png')]>" alt="Sign in with Google">
+              <div class="oppia-signin-google">
+                <i class="oppia-signin-google-icon fa fa-google-plus"></i>
```
This also needs the accessibility fix mentioned previously.

shaz13 replied:
For screen readers? The span element? 
 `<span class="oppia-icon-accessibility-label">Sign In with Google</span>`

anthkris replied:
Yep!

anthkris replied:
@shaz13 Although.... Let me take that back. This actually has text in it. That may not be necessary. I need to test it.

shaz13 replied:
Sure, @anthkris. If its needed then after confirmation will submit a next push.

anthkris replied:
Interestingly, I can't even get to that button from the screenreader (related to another documented issue). So the screenreader doesn't see it at all right now and simply uses the sign in button to sign in.

Everything else LGTM!
